ndindex/booleanarray.py

```diff
@@ -0,0 +1,149 @@
+from numpy import bool_, count_nonzero
+
+from .array import ArrayIndex
+from .ndindex import asshape
+
+class BooleanArray(ArrayIndex):
```

Comment:
Would it be more correct to call this a BooleanArrayIndex?

Reply:
Or even BooleanIndex?  I guess Tuple and Integer don't have "Index" appended.  Maybe it would be most correct to call ArrayIndex just Array

Reply:
Yeah, I haven't been appending "Index" to class names, which may or may not be a bad idea. I did it for ArrayIndex because it's an abstract base class.

Reply:
That's fair